File: LeetCode/6. ZigZag Conversion/Solution.java
public class Solution {
    public static void main(String[] args) {
        String s = "abcdefghijklmnopqrstuvwxyz";
        System.out.println(new Solution().convert(s, 1));
    }
    public String convert(String s, int numRows) {
        StringBuilder[] sb = new StringBuilder[numRows];
        for (int i = 0; i < numRows; i++) {
            sb[i] = new StringBuilder();
        }
        int i = 0;
        while (i < s.length()) {
            for (int j = 0; j < numRows && i < s.length(); j++, i++) {
                sb[j].append(s.charAt(i));
            }
            for (int j = numRows - 2; j > 0 && i < s.length(); j--, i++) {
                sb[j].append(s.charAt(i));
            }
        }

        for (i = 1; i < numRows; i++) {
            sb[0].append(sb[i]);
        }
        return sb[0].toString();

    }
}

File: LeetCode/6. ZigZag Conversion/solution.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        ss = [''] * numRows
        cycle = 2 * numRows - 2
        for idx, ch in enumerate(s):
            idx %= cycle
            if idx < numRows:
                ss[idx] += ch
            else:
                ss[cycle - idx] += ch
        return ''.join(ss)


so = Solution()
print(so.convert("A", 2))
